lib/proxyquire.js

```diff
@@ -161,18 +190,29 @@ Proxyquire.prototype._require = function(module, stubs, path) {
 
   // Only ignore the cache if we have global stubs
   if (this._containsRuntimeGlobal) {
-    return this._withoutCache(module, stubs, path, Module._load.bind(Module, path, module));
+    return this._withoutCache(module, resolvedStubs, path, Module._load.bind(Module, path, module));
   } else {
     return Module._load(path, module);
   }
 };
 
+// `module` here is the module that's invoking proxyquire.
+// `path` is the path of the module we're going to require.
+//
 Proxyquire.prototype._withoutCache = function(module, stubs, path, func) {
   // Temporarily disable the cache - either per-module or globally if we have global stubs
   var restoreCache = this._disableCache(module, path);
+  var resolvedPath = Module._resolveFilename(path, module);
+  var self = this;
+
+  var resolvedStubs = {};
```

Comment:
Try not to assign into objects as a side effect like this. `stubs = Object.keys(stubs).reduce(reducer, {})`.